source/gloperate/source/pipeline/Stage.cpp

```diff
@@ -530,5 +599,51 @@ void Stage::forAllOutputs(std::function<void(gloperate::AbstractSlot *)> callbac
     }
 }
 
+std::uint64_t Stage::lastCPUTime() const
+{
+    return m_lastCPUDuration;
+}
+
+std::uint64_t Stage::lastGPUTime() const
+{
+    return m_lastGPUDuration;
+}
+
+void Stage::setMeasurementFlag(bool flag, bool recursive)
+{
+    m_currentCPUDuration = 0;
+    m_lastCPUDuration = 0;
+    m_lastGPUDuration = 0;
+    m_enableMeasurement = flag;
+
+    if (recursive && isPipeline())
```

Comment:
Please convert this method into a virtual member function on `Stage` and `Pipeline` while handling the substage-iteration on within the `Pipeline`.